admin-dev/themes/new-theme/js/pages/product/components/combination-list/bulk-choices-selector.ts

```diff
@@ -59,21 +87,38 @@ export default class BulkChoicesSelector {
    * Delegated event listener on tabContainer, because every checkbox is re-rendered with dynamic pagination
    */
   private listenCheckboxesChange(): void {
+    this.eventEmitter.on(CombinationEvents.listRendered, () => {
+      const bulkSelectedAll = this.tabContainer
+        .querySelector<HTMLInputElement>(`${CombinationMap.bulkSelectAllCheckboxes}:checked`);
+      this.checkAll(!!bulkSelectedAll);
+    });
+
     this.tabContainer.addEventListener('change', (e) => {
       const checkbox = e.target;
 
       if (!(checkbox instanceof HTMLInputElement)) {
         return;
       }
 
-      const isBulkSelectAll = checkbox.matches(CombinationMap.bulkSelectAll);
+      const isBulkSelectAll = checkbox.matches(`${CombinationMap.bulkSelectAllCheckboxes}`);
 
       // don't proceed if its not one of the expected checkboxes
       if (!isBulkSelectAll && !checkbox.matches(CombinationMap.tableRow.isSelectedCombination)) {
         return;
       }
 
       if (isBulkSelectAll) {
+        const bulkSelectAllCheckboxes = this.tabContainer.querySelectorAll(CombinationMap.bulkSelectAllCheckboxes);
```

Comment:
I took me some time to understand this one I believe the goal here is to keep only one of the "select all" checkboxes checked when one is clicked, so it unchecks the other(s) on right? Just a small comment to explain this would be nice ;)

Reply:
yes, there was a todo comment that explained it all, idk how didn't you see it :smile: But yep, Ill add a comment as now im see there is no better option to do it